test/unit/object/test_task.py

```diff
@@ -0,0 +1,113 @@
+from __future__ import unicode_literals
+
+import json
+import pytest
+
+from mock import Mock
+from boxsdk.config import API
+from boxsdk.object.task import Task
+from boxsdk.object.task_assignment import TaskAssignment
+from boxsdk.network.default_network import DefaultNetworkResponse
+
+
+@pytest.fixture(scope='module')
+def delete_task_response():
+    # pylint:disable=redefined-outer-name
+    mock_network_response = Mock(DefaultNetworkResponse)
+    mock_network_response.ok = True
+    return mock_network_response
+
+
+def test_get(test_task, mock_box_session):
+    expected_url = '{0}/tasks/{1}'.format(API.BASE_API_URL, test_task.object_id)
+    mock_box_session.get.return_value.json.return_value = {
+        'type': test_task.object_type,
+        'id': test_task.object_id,
+        'due_at': '2014-04-03T11:09:43-07:00',
+        'action': 'review',
+        'message': 'Test Message',
+    }
+    retrieved_task = test_task.get()
+    mock_box_session.get.assert_called_once_with(expected_url, headers=None, params=None)
+    assert isinstance(retrieved_task, Task)
+    assert retrieved_task.object_type == test_task.object_type
+    assert retrieved_task.object_id == test_task.object_id
+
+
+def test_update(test_task, mock_box_session):
+    new_message = 'New Message'
+    expected_url = '{0}/tasks/{1}'.format(API.BASE_API_URL, test_task.object_id)
+    mock_box_session.put.return_value.json.return_value = {
+        'type': test_task.object_type,
+        'id': test_task.object_id,
+        'message': new_message,
+    }
+    expected_body = {
+        'message': new_message,
+    }
+    updated_task = test_task.update_info({'message': new_message})
+    mock_box_session.put.assert_called_once_with(expected_url, data=json.dumps(expected_body), headers=None, params=None)
+    assert isinstance(updated_task, Task)
+    assert updated_task.message == new_message
+    assert updated_task.object_type == test_task.object_type
+    assert updated_task.object_id == test_task.object_id
+
+
+def test_delete_policy_return_the_correct_response(
+        test_task,
+        mock_box_session,
+        delete_task_response,
+):
+    # pylint:disable=redefined-outer-name
+    mock_box_session.delete.return_value = delete_task_response
+    response = test_task.delete()
+    # pylint:disable=protected-access
+    expected_url = test_task.get_url()
```

Comment:
This URL should be hardcoded to ensure the test actually checks that the implementation is correct